components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java

```diff
@@ -979,6 +1019,59 @@ private static void buildBasicInformation(IdentityProvider fedIdp, Map<String, S
         }
     }
 
+    private static String handleCertificateDeletion(String oldCertificateFile, String deletedCertificateValues) {
+
+        String decodedOldCertificate = new String(Base64.decode(oldCertificateFile));
+        String decodedDeletedCertificate = new String(Base64.decode(deletedCertificateValues));
+        StringBuilder latestCertificateValue = new StringBuilder();
+
+        return getUpdatedCertificateValue(decodedDeletedCertificate, decodedOldCertificate,
+                latestCertificateValue);
+    }
+
+    private static String handleCertificateAddition(String oldCertFile, String certFile) {
+
+        String decodedOldCertificate = new String(Base64.decode(oldCertFile));
+        String decodedNewCertificate = new String(Base64.decode(certFile));
+
+        // Create the StringBuilder with the old certificates values.
+        StringBuilder finalUpdatedCertificate = new StringBuilder(decodedOldCertificate);
+        return getUpdatedCertificateValue(decodedOldCertificate, decodedNewCertificate,
+                finalUpdatedCertificate);
+    }
+
+    private static String getUpdatedCertificateValue(String decodedCertificate1, String decodedCertificate2,
```

Comment:
what are decodedCertificate1 and decodedCertificate2. can't we have more meaningful names